python/tank/templatekey.py

```diff
@@ -119,7 +150,7 @@ def validate(self, value):
             self._last_error = "%s Illegal value: %s is forbidden for this key." % (self, value)
             return False
 
-        if not((value is None) or (self.choices == [])):
+        if value is not None and self.choices:
```

Comment:
not sure I agree with you on this one as we often make use of the fact an empty list equates to False.  Plus getting the length seems redundant.